janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -252,6 +252,11 @@
                 "this parameter to true but we do recommend to reindex your indexes and don't use this parameter.",
             ConfigOption.Type.MASKABLE, false);
 
+    public static final ConfigOption<Long> CLIENT_KEEP_ALIVE =
+        new ConfigOption<>(ELASTICSEARCH_NS, "client-keep-alive",
+            "Set a keep-alive timeout (in milliseconds)",
+            ConfigOption.Type.GLOBAL_OFFLINE, Long.class);
```

Comment:
I didn't research apache nio http client much but why are you using `ConfigOption.Type.GLOBAL_OFFLINE` here?
Can't we use `ConfigOption.Type.LOCAL` here? Are there possible problems which might arise if we don't use `GLOBAL_OFFLINE`?

Reply:
Yes, I was not sure about the right mutability level to use here. Actually yes we could use `ConfigOption.Type.LOCAL`, the only thing is that if multiple JanusGraph nodes does not use the same keep-alive policy that could lead to weird issues that would be hard to debug. But it is maybe implicit for any conf that it should be the same across all cluster nodes.
I guess you have more experience with these mutability levels, so if you think that `LOCAL` makes more sense, I'll change to that.

Reply:
@debelyoo Could you please describe or refer to problems which you are talking about? I didn't experiment with different keep-alive policies, so I don't know what problems it may cause if different JanusGraph nodes use different keep-alive policies.

Reply:
The problem with these keep-alive policies is that if the keep-alive timeout is longer than the FW rules timeout, the connection between Janus and ES will be cut *sometimes* and will lead to errors while indexing data in mixed indices. If  not all nodes have the same keep-alive policy, these errors will be really sporadic and difficult to reproduce (depending on the node your request is passing by).

Reply:
Good point! Sorry for taking so long to response.